harness/determined/cli/master.py

```diff
@@ -11,64 +11,65 @@
-                )
-            field_mask = ["log"]
-        elif args.level:
-            master_config = bindings.v1Config(log=bindings.v1LogConfig(level=args.level))
-            field_mask = ["log.level"]
-        elif args.color:
-            if args.color.lower() == "on":
-                master_config = bindings.v1Config(log=bindings.v1LogConfig(color=True))
-            elif args.color.lower() == "off":
-                master_config = bindings.v1Config(log=bindings.v1LogConfig(color=False))
-            else:
-                raise CliError(
-                    "Error: Invalid value for --color provided: "
-                    + args.color
-                    + ". This optional parameter only accepts 'on' and 'off' values."
-                )
-            field_mask = ["log.color"]
-        else:
-            raise CliError(
-                "Error: No log level or log color provided. Either log level or log"
-                + "  color is required to make changes to the master config log setting."
-            )
-        req = bindings.v1PatchMasterConfigRequest(
-            config=master_config, fieldMask=bindings.protobufFieldMask(paths=field_mask)
-        )
-        resp = bindings.patch_PatchMasterConfig(cli.setup_session(args), body=req).config
-    elif not args.log and (args.level or args.color):
+    resp = bindings.get_GetMasterConfig(cli.setup_session(args)).config
+    if args.json:
+        render.print_json(resp)
+    else:
+        print(util.yaml_safe_dump(resp, default_flow_style=False))
+
+
+def get_field_mask(args: Namespace) -> List[str]:
+    if args.log_color and args.log_level:
+        return ["log"]
+    elif args.log_color:
+        return ["log.color"]
+    elif args.log_level:
+        return ["log.level"]
+    else:
         raise CliError(
-            "Error: Invalid command: --level and/or --color used without --log. Please try"
-            + " again using 'det master config --log --level <level> --color <on/off>'."
+            "Error: No config values provided, at least one value required to set master config. "
+            + "Currently, --log.level and --log.color are the supported set config attributes."
         )
+
+
+def get_color_value(log_color: str) -> Any:
+    if not log_color:
+        return None
+
+    if log_color.lower() == "on":
+        return True
+    elif log_color.lower() == "off":
+        return False
     else:
-        resp = bindings.get_GetMasterConfig(cli.setup_session(args)).config
+        raise CliError(
+            "Error: Invalid value for --color provided: "
+            + log_color
+            + ". This optional parameter only accepts 'on' and 'off' values."
+        )
+
+
+@authentication.required
+def set_master_config(args: Namespace) -> None:
+    master_config = bindings.v1Config(
+        log=bindings.v1LogConfig(
+            level=get_log_level(args.log_level), color=get_color_value(args.log_color)
+        )
+    )
```

Comment:
it think if the other comments are resolved, this should end up looking something like:

```
    log_config = bindings.v1LogConfig()
    field_masks = []
    if "log_color" in args:
        log_config.color = True if args.log_color == "on" else False
        field_masks.append("log.color")
    if "log_level" in args:
        log_config.level = bindings.v1LogLevel[args.log_level]
        field_masks.append("log.level")

    master_config = bindings.v1Config(log=log_config)
    req = bindings.v1PatchMasterConfigRequest(
        config=master_config, fieldMask=bindings.protobufFieldMask(paths=field_masks)
    )
    
```

Reply:
done